CRM/Utils/Array.php

```diff
@@ -148,7 +148,12 @@ public static function &xml(&$list, $depth = 1, $seperator = "\n") {
       else {
         // make sure we escape value
         $value = self::escapeXML($value);
-        $xml .= "<{$name}>$value</{$name}>{$seperator}";
+        if (is_numeric($name) && is_numeric($value) && ctype_digit(substr($name, 0, 1))) {
+          $xml .= "<id>$value</id>{$seperator}";
```

Comment:
(1) In the case where we have `$list['foo'] == 5`, wouldn't you expect it to output `<foo>5</foo>`? This seems like it's hard-coded to output `<id>5</id>` regardless of field name.

(2) It might make sense to add a test-case to https://github.com/civicrm/civicrm-core/blob/4.7.10/tests/phpunit/CRM/Utils/ArrayTest.php#L7 which lists a few examples of input-arrays and expected-XML. (Note: Jenkins will run tests automatically when you push changes.)

(3) Is this a change in the REST XML format that would break existing consumers?